_posts/2017-01-09-migrating-to-puppet4.md

```diff
@@ -0,0 +1,21 @@
+---
+layout: post
+title: Migrating a Module to Puppet 4
+date: 2017-01-09
+github_username: bastelfreak
+twitter_username: BastelsBlog
+---
+
+
+We announced the deprecation of Puppet 3 in our modules
+[already](https://voxpupuli.org/blog/2016/12/22/putting-down-puppet-3/). There
+are a few steps needed to be taken before a Puppet-4-only release can be done:
+
+* Create a new branch which is called `puppet`
+* Check the current metadata.json for the required Puppet version and bump it if needed
+  * We already have a few modules that only support Puppet 4
+  * The minimal required Puppet Version should not be 4.0.0 but [4.6.1](https://github.com/voxpupuli/community-triage/blob/master/modules/notes/2017-01-05.md#discussion)
+* Check the [.sync.yml](https://github.com/voxpupuli/puppet-tea/blob/e49d6d1ce8ba71c2123edf9fae45cde19e603ec3/.sync.yml#L3-L17), sometimes the `.travis.yml` or `Gemfile` files are unmanaged or modified because the module is already Puppet 4 only. Remove the settings from `.sync.yml`
+* Do a modulesync with at least version XY ([how to do it](https://github.com/voxpupuli/modulesync_config#how-to-use-it))
```

Comment:
xy: The needed changes are already merged in modulesync_config, but unreleased.  https://github.com/voxpupuli/modulesync_config/pull/297